python/revng/internal/cli/_commands/idb_converter.py

```diff
@@ -347,7 +347,17 @@ def _convert_idb_type_to_revng_type(
                 aliased_type, ordinal=aliased_type_ordinal
             )
 
-            resulting_definition = m.TypedefDefinition(Name=type_name, UnderlyingType=underlying)
+            if (
+                aliased_type.get_name() == type_name
+                and type.is_decl_const() == aliased_type.is_decl_const()
+                and isinstance(underlying, m.DefinedType)
+            ):
+                # skip unnecessary typedefs
+                resulting_definition = self.unwrap_definition(underlying)
+            else:
+                resulting_definition = m.TypedefDefinition(
+                    Name=type_name, UnderlyingType=underlying
+                )
```

Comment:
Add a comment about what's going on and why

Reply:
Done:
```
            # IDA uses c-like type system in that for each struct/union a corresponding typedef
            # is introduced. In our model type system, such typedefs are already included
            # in the struct/union definitions (we emit them automatically for each aggregate, with
            # no need for an explicit typedef - de facto, always introducing their names into
            # the global namespace).
            #
            # As such, when importing from IDA, unless a typedef carries any extra information,
            # there's no point of importing it (it's harmful, actually, as it introduces multiple
            # types sharing the same name) - so we just skip it here.
```